From ef420418169d2fb6dde804c94440d37c47aefe5c Mon Sep 17 00:00:00 2001
From: Mark Wielaard <mark@klomp.org>
Date: Fri, 18 Sep 2026 00:24:34 +0200
Subject: [PATCH 50/56] UPSTREAM: stdlib: Don't call clearenv from
 __libc_setenv_freemem

Since commit 7a61e7f557a9 ("stdlib: Make getenv thread-safe in more
cases") clearenv doesn't call any deallocation functions anymore.
__libc_setenv_freemem (called from __libc_freeres) now clears all
backing arrays. So there is no reason anymore to call clearenv from
__libc_setenv_freemem.

Tested against valgrind memcheck with --run-libc-freeres=yes which is
the default.

Reviewed-by: Florian Weimer <fweimer@redhat.com>
(cherry picked from commit b837aae83df8fe80c8977b5ed5c538aebd2b152a)
---
 stdlib/setenv.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/stdlib/setenv.c b/stdlib/setenv.c
index 1f7a4498b0..cbf83f8f8e 100644
--- a/stdlib/setenv.c
+++ b/stdlib/setenv.c
@@ -384,9 +384,6 @@ clearenv (void)
 void
 __libc_setenv_freemem (void)
 {
-  /* Remove all traces.  */
-  clearenv ();
-
   /* Clear all backing arrays.  */
   while (__environ_array_list != NULL)
     {
-- 
2.55.0

